test/sqllogictest/array_fill.slt

```diff
@@ -0,0 +1,379 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Misc. types
+
+query T
+SELECT array_fill(5, array[3])
```

Comment:
Can the first argument legitimately be NULL, to produce an array of NULLs?

Reply:
Yep https://github.com/MaterializeInc/materialize/pull/19694/files#diff-42318d086d07083831f49899dd2b3bef26d2b81e0af737a10ea78f490fcdf82eR188